⛔️ Add detailed error messages to splittable checkout response

daniel-rose opened the pull request:
- effected bundle: fond-of-oryx/splittable-checkout
 - use error messages from checkout response
 - prepare unit tests
 - ...

sourcecube commented on bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/ErpOrderDependencyProvider.php:
```diff
@@ -0,0 +1,282 @@
+     *
+     * @return void
+     */
+    protected function validatePlugin(array $plugins, string $class): void
+    {
+        foreach ($plugins as $plugin) {
+            if (($plugin instanceof $class) === false) {
+                throw new WrongInterfaceException(sprintf(
+                    'Plugin %s has to implement interface from type %s',
+                    get_class($plugin),
+                    $class
+                ));
+            }
+        }
+    }
```
Sweet 👌  .. `InvalidInterfaceException` might be a good fit as well..

sourcecube commented on bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/Propel/Schema/erp_order.schema.xml:
```diff
@@ -0,0 +1,96 @@
+        <column name="canceled_quantity" required="false" type="INTEGER"/>
+        <column name="ordered_quantity" required="false" type="INTEGER"/>
+        <column name="shipped_quantity" required="false" type="INTEGER"/>
+        <column name="invoice_quantity" required="false" type="INTEGER"/>
```
All other fields are past tense except `invoice_quantity` is that as wanted ?

daniel-rose replied:
maybe invoiced_quantity?

sourcecube commented on dandelion.json:
```diff
@@ -13,13 +13,18 @@
             "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/erp-order.git",
             "path": "bundles/erp-order/",
-            "version": "1.0.0"
+            "version": "alpha-0.0.1"
```
I'm not sure but I believe the right way for semantic versioning would be as follow: 

`$major . $minor . $patch - $metalabel` 
Example: `1.0.0-alpha.0` or `1.0.0-alpha.1`

daniel-rose commented on bundles/erp-order-extension/tests/_bootstrap.php:
```diff
@@ -0,0 +1,9 @@
+<?php
+
+$pathToAutoloader = codecept_root_dir('vendor/autoload.php');
+
+if (!file_exists($pathToAutoloader)) {
+    $pathToAutoloader = codecept_root_dir('../../autoload.php');
+}
+
+require_once $pathToAutoloader;
```
```php
<?php

$pathToAutoloader = codecept_root_dir('vendor/autoload.php');

if (!file_exists($pathToAutoloader)) {
    $pathToAutoloader = codecept_root_dir('../../vendor/autoload.php');
}

if (!file_exists($pathToAutoloader)) {
    $pathToAutoloader = codecept_root_dir('../../autoload.php');
}

if (!file_exists($pathToAutoloader)) {
    $pathToAutoloader = codecept_root_dir('../../../../autoload.php');
}

require_once $pathToAutoloader;

```

daniel-rose commented on bundles/erp-order/LICENSE:
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Maybe 2021?

daniel-rose committed:
⛔️ Add detailed error messages to splittable checkout response
 - effected bundle: fond-of-oryx/splittable-checkout
 - use error messages from checkout response
 - prepare unit tests
 - ...
